clients/venice-admin-tool/src/main/java/com/linkedin/venice/ZkCopier.java

```diff
@@ -1,31 +1,69 @@
 package com.linkedin.venice;
 
-import static com.linkedin.venice.zk.VeniceZkPaths.ADMIN_TOPIC_METADATA;
-import static com.linkedin.venice.zk.VeniceZkPaths.EXECUTION_IDS;
-import static com.linkedin.venice.zk.VeniceZkPaths.PARENT_OFFLINE_PUSHES;
-import static com.linkedin.venice.zk.VeniceZkPaths.ROUTERS;
-import static com.linkedin.venice.zk.VeniceZkPaths.STORES;
+import static com.linkedin.venice.zk.VeniceZkPaths.CLUSTER_ZK_PATHS;
 import static com.linkedin.venice.zk.VeniceZkPaths.STORE_CONFIGS;
-import static com.linkedin.venice.zk.VeniceZkPaths.STORE_GRAVEYARD;
 
 import com.linkedin.venice.exceptions.VeniceException;
 import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import org.apache.commons.io.FileUtils;
+import org.apache.helix.zookeeper.datamodel.serializer.ByteArraySerializer;
+import org.apache.helix.zookeeper.impl.client.ZkClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.zookeeper.CreateMode;
 
 
+/**
+ * <p>
+ *   This class contains methods to 1)migrate Venice-specific metadata from a source ZooKeeper (ZK) to a destination ZK
+ *   and 2)extract Venice-specific paths from an input text file containing ZK paths to an output text file.
+ *   We implement a tree data structure to represent ZK paths as nested children and use it to filter out Venice-specific
+ *   paths efficiently.
+ * </p>
+ */
 public class ZkCopier {
-  private static final Set<String> CLUSTER_ZK_PATHS = new HashSet<>(
-      Arrays.asList(ADMIN_TOPIC_METADATA, EXECUTION_IDS, PARENT_OFFLINE_PUSHES, ROUTERS, STORE_GRAVEYARD, STORES));
+  private static final Logger LOGGER = LogManager.getLogger(ZkCopier.class);
+
+  /**
+   * Migrate Venice-specific metadata from a source ZK to a destination ZK.
+   * @param srcZkClient source ZK client
+   * @param destZkClient destination ZK client
+   * @param clusterNames set of cluster names
+   * @param basePath base path for ZK
+   * @Note: {@code destZkClient} should be a fresh ZK server or must not contain any Venice-specific metadata that needs to be migrated,
+   * otherwise a {@code ZkNodeExistsException} will be thrown
+   */
+  public static void migrateVenicePaths(
+      ZkClient srcZkClient,
+      ZkClient destZkClient,
+      Set<String> clusterNames,
+      String basePath) {
+    if (!basePath.startsWith("/")) {
+      throw new VeniceException("Base path must start with a forward slash (/)");
+    }
+    TreeNode srcZkPathsTree = getZkClientPathsTree(srcZkClient, clusterNames, basePath);
+    List<String> destZkPathsList = getVenicePathsFromTree(srcZkPathsTree, clusterNames, basePath);
```

Comment:
There seems to be duplicate logic in `getZkClientPathsTree` and `getVenicePathsFromTree`. `getZkClientPathsTree` is building a tree of only the zk paths that we expect to copy over. With that tree, I don't think we would expect `getVenicePathsFromTree` to return anything different than all the nodes in `srcZkPathsTree`, right?

Reply:
Is building the `srcZkPathsTree` with `getZkClientPathsTree()` and then traversing it again to build the list using `pathsTreeToList()` is inefficient since it is passes through the tree twice?
I wrote a new method `getZkClientPathsList` to directly build the list of paths that we need from the ZkClient.

Reply:
Actually, I would prefer if we reused as much logic between the two as possible. It is harder to maintain two pieces of code that essentially are supposed to do the same thing. Going through the tree twice is okay